src/Microsoft.AspNet.Authentication.OAuth/OAuthHandler.cs

```diff
@@ -89,7 +88,8 @@ public OAuthHandler(HttpClient backchannel)
                 var value = query["error"];
                 if (!StringValues.IsNullOrEmpty(value))
                 {
-                    Logger.LogVerbose("Remote server returned an error: " + Request.QueryString);
+                    Logger.LogWarning("Remote server returned an error: " + value);
```

Comment:
I had expected the new error event to be invoked here where you still had the original error information. That said, it looks like it would be hard to take meaningful action from this point in the call stack, the return value does not give you good control over what happens next. OIDC has some interesting hacks for this issue.